docs/rules/id-blacklist.md

```diff
@@ -0,0 +1,66 @@
+# Blacklist certain identifiers to prevent them being used (id-blacklist)
+
+> "There are only two hard things in Computer Science: cache invalidation and naming things." — Phil Karlton
+
+Bad names can lead to hard to decipher code. Using generic names, such as `data` don't infer much about the code and the values it receives. This rule allows you to configure a blacklist of bad identifier names, that you don't want to see in your code.
+
+## Rule Details
+
+This rule compares assignments and function definitions to a provided list of identifier names. If the identifier is present in the list, it will return an error.
+It doesn't apply to function calls, so that you can still use functions or objects you do not have control over.
+
+### Options
+
+This rule needs a a set of identifier names to blacklist, like so:
+
+```json
+{
+    "rules": {
+        "id-blacklist": [2, "data", "err", "e", "cb", "callback"]
+    }
+}
+```
+
+For the rule in this example, the following patterns are considered problems:
+
+```js
+/*eslint id-blacklist: [2, "data", "err", "e", "cb", "callback"] */
+
+var data = {...};                  /*error Identifier 'data' is blacklisted*/
+
+function callback() {              /*error Identifier 'callback' is blacklisted*/
+    // ...
+}
+
+element.callback = function() {    /*error Identifier 'callback' is blacklisted*/
+    // ...
+};
+
+var itemSet = {
+    data: [...]                    /*error Identifier 'data' is blacklisted*/
+};
+```
+
+The following patterns are not considered problems:
+
+```js
+/*eslint id-blacklist: [2, "data", "err", "e", "cb", "callback"] */
```

Comment:
If I do `callback()` or `foo.callback()` or `bar = foo.callback`, will that warn? If not, can you add those into this list? It would help to clarify when things are flagged.